framework/src/systems/NonlinearEigenSystem.C

```diff
@@ -152,11 +152,13 @@ NonlinearEigenSystem::postAddResidualObject(ResidualObject & object)
   // If it is an eigen kernel, mark its variable as eigen
   if (vtags.find(_Bx_tag) != vtags.end() || mtags.find(_B_tag) != mtags.end())
   {
+    // Note: the object may be on the displaced system
+    auto sys = object.parameters().get<SystemBase *>("_sys");
```

Comment:
A quick comment: we cannot do `object.variable().eigen(true)` because `object.variable()` returns a const reference to the variable. I am not sure whether this code here is the best way to obtain a non-const variable for the residual object. Suggestions are welcome.

Reply:
This is fine. The system owns the variables so it makes sense to modify them through the system APIs (as opposed to through the residual objects)